src/data/boundary-landing.json

```diff
@@ -1,167 +1,85 @@
 {
-  "heading": "Boundary Overview",
```

Comment:
Most of the lines of code change in this PR are content.

For `waypoint-landing.json` & `vault-landing.json`, I've populated with content based on the Figma specs (linked in [🎟️ Implement new designs for Product Landing pages](https://app.asana.com/0/1201903760348480/1201979547919472/f)).

For all other products (including this one, `boundary`), none of the pages are visible on https://developer.hashi-mktg.com, and they won't be visible in the beta either. So I've stubbed in _very_ rough placeholder content based on Vault's content. (Stubbing this because the previous content format is now invalid, since we're refactoring the page).